drivers/frequency/ad9517/ad9517.c

```diff
@@ -287,6 +287,7 @@ int32_t ad9517_read(struct ad9517_dev *dev,
 	int32_t ret = 0;
 	uint8_t tx_buffer[3] = {0, 0, 0};
 	uint8_t i = 0;
+	*reg_value = 0;
```

Comment:
is this really necessary? 
it looks like `*reg_value <<= 8;` does something with the previous reg_value. it might be wrong though, but i'd rather change the implementation.

Reply:
*reg_value <<= 8; was there for multi-byte registers.
And yes, this is really necessary, the reg_value contains junk otherwise in the 3 uppermost bytes (when reading 1 byte registers).